### PR TITLE
add inform and trap PDU types

### PR DESCRIPTION
**Pull request by stakach**
Also provides a helper for sending inform requests

### CODE REVIEW

File: lib/netsnmp/client.rb [hunk 1]
```diff
@@ -135,6 +135,17 @@ def set(*oid_opts)
       values.size > 1 ? values : values.first
     end
 
+    # Perform a SNMP INFORM Request
+    #
+    # @see {NETSNMP::Varbind#new}
+    #
+    def inform(oid_opts)
```
Comment (HoneyryderChuck) [inline lib/netsnmp/client.rb]: Consider keeping the same signature as the other methods (`*oid_opts`).

File: lib/netsnmp/client.rb [hunk 2]
```diff
@@ -135,6 +135,17 @@ def set(*oid_opts)
...
+      request = @session.build_pdu(:inform, oid_opts)
+      response = handle_retries { @session.send(request) }
+      yield response if block_given? 
+      response
```
Comment (HoneyryderChuck) [inline lib/netsnmp/client.rb]: What do you think about keeping here the same pattern as the other methods here? : 

```ruby
values = response.varbinds.map { |v| [v.oid, v.value] }
 values.size > 1 ? values : values.first
```

unless you had your reasons not to.

**Comment by HoneyryderChuck** [comment]
@stakach thx for taking another stab at this. Is there a chance that you could add some trap/inform tests against the snmpsim?

**Comment by stakach** [comment]
In terms of the tests, I'll give that a go

**Comment by HoneyryderChuck** [comment]
@stakach I opted for merging it in, as some other client requests (get_bulk, for example) are not really tested, so there's no point in requiring you to do it. Feel free to merge them in a separate PR though, in case you want to get acquainted with the testing environment.

**Pull request merged**